File: app/src/main/java/com/ytempest/wanandroid/di/module/AllFragmentModule.java
package com.ytempest.wanandroid.di.module;

import com.ytempest.wanandroid.activity.architecture.content.ArchArticleFrag;
import com.ytempest.wanandroid.activity.main.home.HomeFrag;
import com.ytempest.wanandroid.activity.main.knowledge.KnowledgeFrag;
import com.ytempest.wanandroid.activity.main.navigation.NavigationFrag;
import com.ytempest.wanandroid.activity.main.project.ProjectFrag;
import com.ytempest.wanandroid.activity.main.project.content.ProjectContentFrag;
import com.ytempest.wanandroid.di.component.BaseFragmentComponent;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * @author heqidu
 * @since 2020/6/30
 */
@Module(subcomponents = BaseFragmentComponent.class)
public abstract class AllFragmentModule {

    @ContributesAndroidInjector(modules = FragmentModules.EmptyModule.class)
    abstract HomeFrag contributeHomeFragInjector();

    @ContributesAndroidInjector(modules = FragmentModules.EmptyModule.class)
    abstract KnowledgeFrag contributeKnowledgeFragInjector();

    @ContributesAndroidInjector(modules = FragmentModules.EmptyModule.class)
    abstract NavigationFrag contributeNavigationFragInjector();

    @ContributesAndroidInjector(modules = FragmentModules.EmptyModule.class)
    abstract ProjectFrag contributeProjectFragInjector();

    @ContributesAndroidInjector(modules = FragmentModules.EmptyModule.class)
    abstract ProjectContentFrag contributeClassifyContentFragInjector();

    @ContributesAndroidInjector(modules = FragmentModules.EmptyModule.class)
    abstract ArchArticleFrag contributeArchArticleFragInjector();
}

File: app/src/main/java/com/ytempest/wanandroid/di/module/AllFragmentModule.kt
package com.ytempest.wanandroid.di.module

import com.ytempest.wanandroid.activity.main.home.HomeFrag
import com.ytempest.wanandroid.activity.main.knowledge.KnowledgeFrag
import com.ytempest.wanandroid.activity.main.navigation.NavigationFrag
import com.ytempest.wanandroid.activity.main.project.ProjectFrag
import com.ytempest.wanandroid.activity.main.project.content.ProjectContentFrag
import com.ytempest.wanandroid.di.component.BaseFragmentComponent
import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * @author heqidu
 * @since 21-2-7
 */
@Module(subcomponents = [BaseFragmentComponent::class])
abstract class AllFragmentModule {
    @ContributesAndroidInjector(modules = [FragEmptyModule::class])
    abstract fun contributeHomeFragInjector(): HomeFrag

    @ContributesAndroidInjector(modules = [FragEmptyModule::class])
    abstract fun contributeKnowledgeFragInjector(): KnowledgeFrag

    @ContributesAndroidInjector(modules = [FragEmptyModule::class])
    abstract fun contributeNavigationFragInjector(): NavigationFrag

    @ContributesAndroidInjector(modules = [FragEmptyModule::class])
    abstract fun contributeProjectFragInjector(): ProjectFrag

    @ContributesAndroidInjector(modules = [FragEmptyModule::class])
    abstract fun contributeClassifyContentFragInjector(): ProjectContentFrag
}